addons/mail/controllers/discuss/public_page.py

```diff
@@ -93,10 +98,13 @@ def _response_discuss_channel_invitation(self, store, channel):
         with replace_exceptions(UserError, by=NotFound()):
             # sudo: mail.guest - creating a guest and its member inside a channel of which they have the token
             __, guest = channel.sudo()._find_or_create_persona_for_channel(
-                guest_name=_("Guest"),
+                guest_name=guest_email if guest_email else _("Guest"),
```

Comment:
I'm not exactly sure when that happens, but the name provided on the welcome page is the email by default, but if we type something else, it still gets replaced by the email.

Reply:
That's actually a bug that was probably introduced when we decided to create guests before the welcome page, rather than after. I'll fix this in the correct version in another PR.